work_queue/src/work_queue_resources.h

```diff
@@ -19,6 +19,7 @@ struct work_queue_resource {
 
 struct work_queue_resources {
 	int64_t tag;                       // Identifies the resource snapshot.
+	time_t time_left;			   	   // Marks the time left for work queue workers
```

Comment:
I think it would be easier to have a end_time in the work_queue_worker.c, rather than a resource field here. time_left behaves differently than the other resources anyway.
With this, in wq worker we compute end_time as time() + time specified from the command line. (in seconds?)